File: problems/1673/solution.py
class Solution:
    def mostCompetitive(self, nums: List[int], k: int) -> List[int]:
        q, count = [], len(nums) - k
        
        for num in nums:
            while q and num < q[-1] and count > 0:
                q.pop()
                count -= 1
            
            q.append(num)
        
        return q[0:k]

File: problems/1673/solution.cpp
class Solution {
public:
    vector<int> mostCompetitive(vector<int>& nums, int k) {
        vector<int> ans;
        deque<int> q;
        int count = nums.size() - k;
        
        for (int i = 0; i < nums.size(); i++) {
            while (!q.empty() && q.back() > nums[i] && count > 0) {
                q.pop_back();
                count--;
            }
            
            q.push_back(nums[i]);
        }
        
        for (int i = 0; i < k; i++) {
            ans.push_back(q.front());
            q.pop_front();
        }
        
        return ans;
    }
};